src/box.tsx

```diff
@@ -21,7 +21,10 @@ Box.displayName = 'Box'
 
 Box.propTypes = {
   ...propTypes,
-  innerRef: PropTypes.func,
+  innerRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.elementType })
```

Comment:
@tusbar thinking something like this would work?

Reply:
Shouldn’t it be `{current: PropTypes.element}`?

Reply:
Won't that exclude other components like `MyComponent` as a ref?

Reply:
Correct me if I’m wrong but isn’t the difference between element/elementType the same between instance/class?

`<Foo />` is an `element` and `Foo` is an `elementType`?

I’ve never used `PropTypes.elementType` before but that’s what I understand from the doc https://reactjs.org/docs/typechecking-with-proptypes.html (which is not really explaining much).

Reply:
https://github.com/facebook/react/tree/master/packages/react-is#determining-an-elements-type